src/plugins/intel_cpu/src/mkldnn_exec_network.cpp

```diff
@@ -77,6 +78,23 @@ MKLDNNExecNetwork::MKLDNNExecNetwork(const InferenceEngine::CNNNetwork &network,
         }
     }
 
+    // WA for inference dynamic batch cases in new API
+    if (isNewApi) {
+        int64_t maxBatchSize = -1;
+        if (canBeExecViaLegacyDynBatch(function, maxBatchSize)) {
+            IE_ASSERT(maxBatchSize > -1);
+            _cfg.canBeExecAsDynBatch = true;
+            upperBoundModel = ngraph::clone_function(*function);
+            std::map<ov::Output<ov::Node>, ov::PartialShape> newInShape;
+            for (const auto& in : upperBoundModel->get_parameters()) {
+                auto newShape = in->get_output_partial_shape(0);
+                newShape[0] = maxBatchSize;
+                newInShape[in] = newShape;
+            }
+            upperBoundModel->reshape(newInShape);
```

Comment:
To be honest I don't like the idea to extend CreateGraph API with upperBound model. Can we perform the same reshape on usual `function`? 
The same story about newly added canBeExecAsDynBatch key. Cannot we just reuse legacy `batchLimit` for that?